### ISSUE TITLE
Scala 2.13

### ISSUE THREAD

**Issue by btmcnellis** [sangria-graphql/sangria#440]
With the Scala 2.13.0-RC1 release, it seems like Scala 2.13 will be here relatively soon. What's the plan for supporting 2.13 in Sangria? I would be willing to help with the migration, though I'm far from an expert on the internals of this library.

**Comment by btmcnellis** [comment]
@OlegIlyenko Do you have any updates on this? Like I said, I'm willing to help with this, and I'm sure other users are too. But it would be nice to have some idea of what you're thinking so users can start to plan ahead.

**Comment by ingemaradahl** [comment]
I've started looking at bringing Sangria to 2.13. First off all of Sangria's dependencies needs to be met for 2.13. The good news is that Oleg kept the list of dependencies small; the only "external" dependency not related to tests is parboiled, which is published for 2.13.0-RC1 (so a proper 2.13 release should be imminent). This leaves the in-project dependencies and test dependencies. Furthermore marshalling is split out into glue packages which also needs to be updated, but that can be done independently of the main package.

So far I've issued a PR for `sangria-marshalling-api` in https://github.com/sangria-graphql/sangria-marshalling-api/pull/4, and `macro-visit` in https://github.com/sangria-graphql/macro-visit/pull/3.

This isn't much but it's something in the right direction. It'd be a shame if Sangria were to fall behind, it's a fantastic project created by a genius developer. I think it's time we all start pitching in.

**Comment by schrepfler** [comment]
Possibly parboiled could be replaced by fastparse?

**Comment by ingemaradahl** [comment]
I'd deem replacing parboiled with something else as an unnecessary risk when the problem most likely will resolve itself. In the interim Sangria could probably be built against parboiled-2.1.6 built against Scala 2.13.0-RC1, as done in https://github.com/jrudolph/json-lenses/pull/32/commits/c01c325476f90fe0423bd5c787807ffaa03ebefb.

https://github.com/jcazevedo/moultingyaml (a test dependency) however looks like an abandoned project which probably won't see a Scala 2.13 release.

**Comment by jazott3** [comment]
I am willing to pitch in here, although I'm not familiar enough with GraphQL to make any major changes. I took a look a few weeks ago at migrating some of the smaller in-project dependencies as well. I don't think I got very far (sangira-marshalling-api, sangria-marshalling-testkit), but this weekend I will review and see if I can't bring  a couple of other subprojects up to speed

**Comment by ingemaradahl** [comment]
That'd be great! I'm not really familiar with the code base either, but moving to 2.13 shouldn't be too complex. I just did a test compile of the main project and I think we're in an OK spot: 3227 warnings (of which I'm guessing the bulk is deprecated Unicode ligatures) and 65 errors, so not too bad.

**Comment by schrepfler** [comment]
I saw some changes to parboiled committed by @xuwei-k 8hrs ago, maybe he's looking at publishing it for 2.13.

**Comment by jazott3** [comment]
I've opened PR (https://github.com/sangria-graphql/sangria-marshalling-testkit/pull/2), built against the open sangria-marshalling-api PR.

**Comment by jazott3** [comment]
I've added a PR for sangria-spray-json (https://github.com/sangria-graphql/sangria-spray-json/pull/1), and it appears there is a PR for sangria-streaming-api (https://github.com/sangria-graphql/sangria-streaming-api/pull/2).

From what I can tell, the remaining dependencies are:
- sangria-argonaut
- sangria-ion
- sangria-monix
- sangria-rxscala

**Comment by jtjeferreira** [comment]
I pushed a PR with my initial work on this. I commented the test dependencies for now, and using the locally published versions of the above mentioned PRs sources are now compiling.

Tests will be a bit more tricky as more dependencies are involved...

**Comment by pjfanning** [comment]
I have the sangria-playground sample working on scala 2.13.0 based on using some snapshot jars that I published. (https://github.com/pjfanning/sangria-playground/tree/scala-2.13)

**Comment by travisbrown** [comment]
The following modules are now available for Scala 2.13:

* https://github.com/sangria-graphql-org/macro-visit/releases/tag/v0.1.2
* https://github.com/sangria-graphql-org/sangria-argonaut/releases/tag/v1.0.1
* https://github.com/sangria-graphql-org/sangria-circe/releases/tag/v1.3.0
* https://github.com/sangria-graphql-org/sangria-ion/releases/tag/v2.0.0
* https://github.com/sangria-graphql-org/sangria-marshalling-api/releases/tag/v1.0.4
* https://github.com/sangria-graphql-org/sangria-marshalling-testkit/releases/tag/v1.0.2
* https://github.com/sangria-graphql-org/sangria-monix/releases/tag/v2.0.0
* https://github.com/sangria-graphql-org/sangria-spray-json/releases/tag/v1.0.2
* https://github.com/sangria-graphql-org/sangria-streaming-api/releases/tag/v1.0.1

I've also published a milestone release of the core module with 2.13 support:

* https://github.com/sangria-graphql-org/sangria/releases/tag/v2.0.0-M1

The rest still need 2.13 PRs:

* https://github.com/sangria-graphql-org/sangria-akka-streams
* https://github.com/sangria-graphql-org/sangria-json4s-jackson
* https://github.com/sangria-graphql-org/sangria-json4s-native
* https://github.com/sangria-graphql-org/sangria-msgpack
* https://github.com/sangria-graphql-org/sangria-play-json
* https://github.com/sangria-graphql-org/sangria-relay
* https://github.com/sangria-graphql-org/sangria-rxscala
* https://github.com/sangria-graphql-org/sangria-slowlog

**Comment by mjadczak** [comment]
I'm working on the sangria-akka-streams PR right now (see also sangria-graphql/sangria-akka-streams#2 )

**Comment by pjfanning** [comment]
I have added PRs for
* https://github.com/sangria-graphql-org/sangria-json4s-jackson
* https://github.com/sangria-graphql-org/sangria-json4s-native
* https://github.com/sangria-graphql-org/sangria-play-json

**Comment by travisbrown** [comment]
Thanks @pjfanning! Any chance you'd want to go ahead and add sbt-release and sbt-mima to those as well?

**Comment by pjfanning** [comment]
@travisbrown I added the mima-check on sangria-play-json and it fails - any suggestion appreciated.

The 2 json4s projects are sbt 0.13 based so it might take a bit longer to upgrade their sbt plugins.

It might take me a few days to get back to the PRs.

**Comment by travisbrown** [comment]
@pjfanning Thanks! And it's definitely fine if MiMa fails, if the breakage is necessary or intentional—the bincompat report is useful info for setting the next release version and writing the release notes.

**Comment by yanns** [comment]
https://github.com/sangria-graphql-org/sangria-json4s-jackson is done

**Comment by yanns** [comment]
https://github.com/sangria-graphql-org/sangria-json4s-native/releases/tag/v1.0.1 is done

**Comment by yanns** [comment]
https://github.com/sangria-graphql-org/sangria-slowlog/releases/tag/v2.0.0-M1

**Comment by travisbrown** [comment]
@yanns I'm publishing sangria-msgpack 2.0.0 now (I updated it a few days ago but haven't published it yet). We're still waiting on RxScala to support 2.13 to publish sangria-rxscala (which is otherwise updated, without needing any significant changes).

I could go ahead and update sangria-relay now, unless you've already started? That's the last.

**Comment by yanns** [comment]
I have not started sangria-relay. You can pick it. Thx! ❤️

**Comment by travisbrown** [comment]
@yanns Great! sangria-msgpack is published and here's the sangria-relay PR: https://github.com/sangria-graphql-org/sangria-relay/pull/6

**Comment by nickhudkins** [comment]
I just checked, and all of the packages support 2.13 now 🎉
